static/js/biomedical/disease/chart.ts

```diff
@@ -247,3 +252,120 @@ export function drawDiseaseSymptomAssociationChart(
       (d) => `Symptom: ${d.name}<br>Odds Ratio Association: ${d.oddsRatio}`
     );
 }
+/**
+ * Draws the disease ontology hierarchy chart for the disease of interest
+ * @param id the div id where the chart is rendered on the page
+ * @param data the disease data passed into the function
+ */
+export function drawDiseaseOntologyHierarchy(
+  id: string,
+  data: DiseaseParentData[]
+): void {
+  // checks if the data is empty or not
+  if (_.isEmpty(data)) {
+    return;
+  }
+  const height = GRAPH_HEIGHT - MARGIN.top - MARGIN.bottom;
+  const width = GRAPH_WIDTH - MARGIN.left - MARGIN.right;
+  const svg = d3
+    .select(`#${id}`)
+    .append("svg")
+    .attr("width", width + MARGIN.left + MARGIN.right)
+    .attr("height", height + MARGIN.top + MARGIN.bottom)
+    .append("g")
+    .attr("transform", "translate(" + MARGIN.left + "," + MARGIN.top + ")");
+  // plots the axes
+  const x = d3
+    .scaleBand()
+    .range([0, width])
+    .domain(
+      data.map((d) => {
+        return d.name;
+      })
+    )
+    .padding(1);
+  svg
+    .append("g")
+    .attr("transform", "translate(0," + (height - MARGIN.bottom) + ")")
+    .call(d3.axisBottom(x))
+    .selectAll("text")
+    .attr("dx", "-.8em")
+    .attr("dy", ".8em")
+    .attr("transform", "rotate(-40)")
+    .style("text-anchor", "end");
+  // color scale for disease nodes
+  const color_scale = d3
+    .scaleSequential<string>()
+    .domain([0, 10])
+    .interpolator(d3.interpolateOrRd);
+  const circleIDFunc = getElementIDFunc(id, "circle");
+  // the circles
+  svg
+    .selectAll("disease-parent-circle")
+    .data(data)
+    .enter()
+    .append("circle")
+    .attr("cx", (d) => {
+      return x(d.name);
+    })
+    .attr("cy", () => {
+      return height - MARGIN.bottom;
+    })
+    .attr("r", (d, i) => {
+      return LEGEND_CIRCLE_RADIUS + DISEASE_PARENT_LEGEND_INCREMENT * i;
+    })
+    .attr("id", (d, i) => circleIDFunc(i))
+    .style("fill", (d, i) => {
+      return color_scale(i + DISEASE_PARENT_LEGEND_INCREMENT);
+    })
+    .call(
+      handleMouseEvents,
+      circleIDFunc,
+      (d) => `Disease Name: ${d.name}<br>Disease Dcid: ${d.dcid}`
+    );
+  // the legend
+  const defs = svg.append("defs");
+  const linearGradient = defs
+    .append("linearGradient")
+    .attr("id", "linear-gradient");
+  linearGradient
+    .attr("x1", "0%")
+    .attr("y1", "0%")
+    .attr("x2", "100%")
+    .attr("y2", "0%");
+  //Set the color for the start (0%)
+  linearGradient
+    .append("stop")
+    .attr("offset", "0%")
+    .attr("stop-color", "#ffa474"); //light orange
```

Comment:
make these colors global constants. But also, shouldn't the start color be the color on the colorScale for the current node? & the end color should be the color on the colorScale for the top parent node?